scripts/github-changelog.cr

```diff
@@ -163,11 +165,17 @@ sections = array.group_by { |pr|
   end || "Other"
 }
 
-titles = sections.keys.sort!
+titles = [] of String
+["Compiler", "Language", "Standard Library", "Tools", "Other"].each do |main_section|
+  titles.concat sections.each_key.select(&.starts_with?(main_section)).to_a.sort!
+end
+sections.keys.sort!.each do |section|
+  titles << section unless titles.includes?(section)
+end
 last_title1 = nil
 
 titles.each do |title|
-  prs = sections[title]
+  prs = sections[title]? || next
```

Comment:
```suggestion
  next unless prs = sections[title]?
```

Reply:
No. I think assignments in trailing conditionals are a code smell.

Reply:
And `next` in assignments is fine with you? What's the logic here?

Reply:
```suggestion
  prs = sections[title]?
  next unless prs
```
?

Reply:
Yes, I think that's fine. It's handling a null state and breaks from the current iteration as a result. That's an established idiom, similar to `x || return` and `x || raise` etc. of which you can find many instances in stdlib, for example.
Separating the expressions is also fine, of course. I just don't see it's necessary.
Unless there's reasonable concern about the readability, I'd like to avoid useless style discussions.

Reply:
Using `next` / `return` or `raise` **within** the assignments should be considered as a code smell.

> That's an established idiom, similar to `x || return` and `x || raise` etc. of which you can find many instances in stdlib, for example.

No, it's not. In stdlib is used only in **5** places (not counting specs) - see https://cs.github.com/crystal-lang/crystal?q=%2F%5Cs%3D%28.%2B%3F%29+%5C%7C%5C%7C+%28next%7Creturn%7Cbreak%29%5CW%2F+-path%3A%2Fspec%2F

> Unless there's reasonable concern about the readability, I'd like to avoid useless style discussions.

_Useless_ for you perhaps, so please keep your judgements for yourself.

Reply:
Sorry, I didn't mean to pass any judgement. Useless discussions are useless, that's a fact. Whether this particular style discussion is useless depends on the condition of whether there's reasonable argument.

Your link doesn't work for me, and I'm sure its results are incomplete. A local code search shows at least 80 instances of `||` plus trailing break expression in `src`, a quarter of them in assignments. So I'm pretty sure this can be considered established. If you'd like to change that, please start a separate discussion. But we should refrain from discussing general style issues in unrelated PRs.